### PR TITLE
Update data model to improve chaining enhancements

### PR DESCRIPTION
**Pull request by sanfern**
This PR updates members of BPFProgram to store eBPF program kernel object filename and prorgam name.

### CODE REVIEW

File: config/config.go [hunk 1]
```diff
@@ -44,19 +44,25 @@ type Config struct {
...
-	XDPRootProgramMapName  string
-	XDPRootProgramCommand  string
-	XDPRootProgramVersion  string
+	XDPRootProgramName       string
+	XDPRootProgramArtifact   string
+	XDPRootProgramMapName    string
+	XDPRootProgramCommand    string
+	XDPRootProgramVersion    string
+	XDPRootProgramObjectFile string
+	XDPRootProgramProgName   string
```
Comment (dthaler) [inline config/config.go:53]: What is the difference between "ProgramName" and "ProgramProgName"?  These names sound synonymous.

File: models/l3afd.go [hunk 2]
```diff
@@ -50,6 +50,8 @@ type BPFProgram struct {
 	ConfigArgs        L3afDNFArgs         `json:"config_args"`           // Map of arguments to config command
 	MonitorMaps       []L3afDNFMetricsMap `json:"monitor_maps"`          // Metrics BPF maps
 	EPRURL            string              `json:"ebpf_package_repo_url"` // Download url for Program
+	ObjectFile        string              `json:"object_file"`           // Object file contains kernel code
+	ProgramName       string              `json:"program_name"`          // BPF program name to load
```
Comment (dthaler) [inline models/l3afd.go:54]: Line 27 has "Name of the BPF program".  What's the difference between that and "BPF program name to load"?

**Review by dthaler — DISMISSED** [review]

File: config/config.go (hunk 1 above)
Comment (sanfern) [reply config/config.go:53]: Here eBPF Program package name was referred to as ProgramName. The ProgramProgName refers to the name of the program that is being loaded.

Comment (dthaler) [reply config/config.go:53]: Too confusing.  Please fix the names to be distinct.   Neither use the word "Package" at present.

**Review by dthaler — APPROVED** [review]

**Review by dthaler — APPROVED** [review]

**Review by jniesz — APPROVED** [review]
LGTM and removed the breaking-change label as we now handle both configs.

**Review by dthaler — APPROVED** [review]

**Pull request merged**